src/main.rs

```diff
@@ -484,30 +532,37 @@ impl BFInterp {
         }
     }
 
-    fn sext(&mut self, op: &Operand, to_type: &TypeRef, dest: &name::Name) {
+    fn szext(&mut self, op: &Operand, to_type: &TypeRef, dest: &name::Name) {
         match op {
             LocalOperand { name, ty } => match ty.as_ref() {
-                Type::IntegerType { bits } => match bits {
-                    32 => match to_type.as_ref() {
-                        Type::IntegerType { bits } => {
-                            match bits {
-                                64 => {
-                                    let constant = Constant::Int {
-                                        bits: 64,
-                                        value: self.get_int_op(self.vars.get(name).unwrap()) as i32
-                                            as i64
-                                            as u64,
-                                    };
-                                    self.vars.insert(
-                                        dest.clone(),
-                                        ConstantOperand(ConstantRef::new(constant)),
-                                    );
-                                }
-                                _ => todo!(),
-                            };
-                        }
-                        _ => todo!(),
-                    },
+                Type::IntegerType { bits: _ } => match to_type.as_ref() {
+                    Type::IntegerType { bits } => {
+                        match bits {
+                            32 => {
+                                let constant = Constant::Int {
+                                    bits: 32,
+                                    value: self.get_int_op(self.vars.get(name).unwrap()) as i32
+                                        as u64,
+                                };
+                                self.vars.insert(
+                                    dest.clone(),
+                                    ConstantOperand(ConstantRef::new(constant)),
+                                );
+                            }
+                            64 => {
+                                let constant = Constant::Int {
+                                    bits: 64,
+                                    value: self.get_int_op(self.vars.get(name).unwrap()) as i64
```

Comment:
Doesn't `get_int_op` already return a `u64`? If so there's really no need for those casts.

Reply:
Yes you're right.